### ISSUE TITLE
thermalctld no longer adds 'speed_tolerance' to the Redis database

### ISSUE THREAD

**Issue by Yagami-Jiang** [sonic-net/sonic-platform-daemons#395]
Hi, i want to know why we should delete 'speed_tolerance' field?
Cause as i know, the command 'show system-health summary' will call the file 'hardware_checker.py' which is at master branch, and it will get the 'speed_tolerance' from Redis database.
If we delete the 'speed_tolerance', an error will be generated when using the command 'show system-health summary'

![image](https://github.com/sonic-net/sonic-platform-daemons/assets/119273528/8929252c-d0e8-40c0-acbc-8533c167d656)

	root@sonic:/home/admin# show system-health summary

	System status summary

	System status LED  amber

	Services:

	Status: OK

	Hardware:

	Status: Not OK

	Reasons: Failed to get speed tolerance for Fantray7_2

		 Failed to get speed tolerance for Fantray7_1

		 Failed to get speed tolerance for Fantray6_2

		 Failed to get speed tolerance for Fantray6_1

		 Failed to get speed tolerance for Fantray5_2

		 Failed to get speed tolerance for Fantray5_1

		 Failed to get speed tolerance for Fantray4_2

		 Failed to get speed tolerance for Fantray4_1

		 Failed to get speed tolerance for Fantray3_2

		 Failed to get speed tolerance for Fantray3_1

		 Failed to get speed tolerance for Fantray2_2

		 Failed to get speed tolerance for Fantray2_1

		 Failed to get speed tolerance for Fantray1_2

		 Failed to get speed tolerance for Fantray1_1

	root@sonic:/home/admin# show platform fan

	  Drawer    LED         FAN    Speed    Direction    Presence    Status          Timestamp

	--------  -----  ----------  -------  -----------  ----------  --------  -----------------

	Fantray1  green  Fantray1_1      56%       INTAKE     Present        OK  20221222 15:29:07

	Fantray1  green  Fantray1_2      56%       INTAKE     Present        OK  20221222 15:29:07

	Fantray2  green  Fantray2_1      55%       INTAKE     Present        OK  20221222 15:29:07

	Fantray2  green  Fantray2_2      56%       INTAKE     Present        OK  20221222 15:29:07

	Fantray3  green  Fantray3_1      55%       INTAKE     Present        OK  20221222 15:29:07

	Fantray3  green  Fantray3_2      56%       INTAKE     Present        OK  20221222 15:29:07

	Fantray4  green  Fantray4_1      55%       INTAKE     Present        OK  20221222 15:29:07

	Fantray4  green  Fantray4_2      56%       INTAKE     Present        OK  20221222 15:29:07

	Fantray5  green  Fantray5_1      56%       INTAKE     Present        OK  20221222 15:29:07

	Fantray5  green  Fantray5_2      56%       INTAKE     Present        OK  20221222 15:29:07

	Fantray6  green  Fantray6_1      56%       INTAKE     Present        OK  20221222 15:29:07

	Fantray6  green  Fantray6_2      56%       INTAKE     Present        OK  20221222 15:29:07

	Fantray7  green  Fantray7_1      55%       INTAKE     Present        OK  20221222 15:29:07

	Fantray7  green  Fantray7_2      56%       INTAKE     Present        OK  20221222 15:29:07

	 N/A    N/A   PSU1_FAN1      29%       INTAKE     Present        OK  20221222 15:29:07

	 N/A    N/A   PSU2_FAN1      29%       INTAKE     Present        OK  20221222 15:29:07

**Comment by Yagami-Jiang** [comment]
@spilkey-cisco Could you help me to fix this issue?

**Comment by spilkey-cisco** [comment]
Speed tolerance is no longer a required field or method for vendors to implement, as it has been replaced by is_under_speed and is_over_speed. health_checker.py should be updated to use these new APIs instead of hard-coding calculations for speed_min_th and speed_max_th.

**Comment by spilkey-cisco** [comment]
I can open a PR for this shortly.

**Comment by Yagami-Jiang** [comment]
> I can open a PR for this shortly.

Sorry, do you mean you will open a PR for health_checker.py , right?

**Comment by spilkey-cisco** [comment]
> > I can open a PR for this shortly.
> 
> Sorry, do you mean you will open a PR for health_checker.py , right?

Do you mean `hardware_checker.py`, as that's where the issue was reported? I will open a PR to fix `hardware_checker.py`.

**Comment by Yagami-Jiang** [comment]
> > > I can open a PR for this shortly.
> > 
> > 
> > Sorry, do you mean you will open a PR for health_checker.py , right?
> 
> Do you mean `hardware_checker.py`, as that's where the issue was reported? I will open a PR to fix `hardware_checker.py`.

Yes. So， please let me know if you open a PR to fix `hardware_checker.py`. Very Thx

**Comment by spilkey-cisco** [comment]
@Yagami-Jiang https://github.com/sonic-net/sonic-buildimage/pull/16689
